internal/controller/nimcache_controller.go

```diff
@@ -495,56 +495,57 @@ func getSelectedProfiles(nimCache *appsv1alpha1.NIMCache) ([]string, error) {
 	return nil, nil
 }
 
-func (r *NIMCacheReconciler) reconcileModelSelection(ctx context.Context, nimCache *appsv1alpha1.NIMCache) (requeue bool, err error) {
+func (r *NIMCacheReconciler) reconcileModelManifestAndModelSelection(ctx context.Context, nimCache *appsv1alpha1.NIMCache) (requeue bool, err error) {
 	logger := r.GetLogger()
 
-	// reconcile model selection pod
-	if isModelSelectionRequired(nimCache) && !isModelSelectionDone(nimCache) {
-		// Create a temporary pod for parsing model manifest
-		pod := constructPodSpec(nimCache)
-		// Add nimCache as owner for watching on status change
-		if err := controllerutil.SetControllerReference(nimCache, pod, r.GetScheme()); err != nil {
-			return false, err
-		}
-		err := r.createPod(ctx, pod)
-		if err != nil {
-			logger.Error(err, "failed to create", "pod", pod.Name)
-			return false, err
-		}
+	// Create a temporary pod for parsing model manifest
+	pod := constructPodSpec(nimCache)
```

Comment:
we should not run the pod on every reconciliation, but only when the configmap is not already created.